funnel/static/js/scripts.js

```diff
@@ -1,80 +1,299 @@
+              queue.dequeue(participantID);
+            }
+            else {
+              $('#' + participantID).find('.js-loader').show();
+            }
+          }
+        });
+      }
     };
-    $(selector).change(handler);
-    $(selector).click(handler);
+  };
 }
 
-function commentsInit(pageURL) {
-    $(".comments .collapse").click(function() {
-        $(this).addClass('hidden');
-        $(this).parent().find('.uncollapse').removeClass('hidden');
-        $(this).parent().parent().find('.com-body').slideUp("fast");
-        $(this).parent().parent().parent().find('.com-children').slideUp("fast");
-        return false;
-    });
+window.Talkfunnel.ParticipantTable = {
+  init: function(config, checkinQ, cancelcheckinQ) {
+    this.Config = {
+      checkinUrl: config.checkinUrl,
+      participantlistUrl: config.participantlistUrl
+    };
+    if (Modernizr.localstorage) {
+      this.checkinQ = checkinQ;
+      this.cancelcheckinQ = cancelcheckinQ;
+    }
 
-    $(".comments .uncollapse").click(function() {
-        $(this).addClass('hidden');
-        $(this).parent().find('.collapse').removeClass('hidden');
-        $(this).parent().parent().find('.com-body').slideDown("fast");
-        $(this).parent().parent().parent().find('.com-children').slideDown("fast");
-        return false;
-    });
+    Ractive.DEBUG = false;
 
-    $(".comments .comment-reply").click(function() {
-        var cfooter = $(this).parent();
-        $('#newcomment input[name="parent_id"]').val(cfooter.attr('data-id'));
-        $('#newcomment input[name="comment_edit_id"]').val('');
-        $("#toplevel-comment").removeClass('hidden');
-        $("#comment-submit").val("Reply"); // i18n gotcha
-        cfooter.after($("#newcomment"));
-        $("#newcomment textarea").focus();
-        return false;
+    this.participantTableContent = new Ractive({
+      el: '#participants-table-content',
+      template: '#participant-row',
+      data: {
+        participants: '',
+        getCsrfToken: function() {
+          return $('meta[name="csrf-token"]').attr('content');
+        },
+        getBadgeUrl: function(pid) {
+          return config.badgeUrl.replace('participant-id',pid);
+        },
+        getEditUrl: function(pid) {
+          return config.editUrl.replace('participant-id',pid);
+        },
+        getCheckinUrl: function() {
+          return config.checkinUrl;
+        }
+      }
+    });
+  },
+  update: function() {
+    var participantTable = this;
+    
+    $.ajax({
+      type: 'GET',
+      url: participantTable.Config.participantlistUrl,
+      timeout: 5000,
+      dataType: 'json',
+      success: function(data) {
+        participantTable.participantTableContent.set('participants', data.participants).then(function() {
+          $('.js-loader').hide();
+          $('.js-total').text(data.total_participants);
+          $('.js-totalcheckin').text(data.total_checkedin);
```

Comment:
`js-total` and `js-totalcheckin` can automatcally update if the data is updated, right? Why do we need to reset it with jQuery?